tests/acceptance/github-release-test.js

```diff
@@ -0,0 +1,99 @@
+import { test } from 'qunit';
+import moduleForAcceptance from '../../tests/helpers/module-for-acceptance';
+import startApp from 'dummy/tests/helpers/start-app';
+import Pretender from 'pretender';
+import Ember from 'ember';
+
+const { run } = Ember;
+
+let server, app, container, store;
+
+moduleForAcceptance('Acceptance | github release', {
+  beforeEach() {
+    server = new Pretender();
+    server.prepareBody = function (body) {
+      return JSON.stringify(body);
+    };
+    app = startApp();
+    container = app.__container__;
+    store = run(container, 'lookup', 'service:store');
+  },
+
+  afterEach() {
+    server.shutdown();
+    run(app, app.destroy);
+    Ember.BOOTED = false;
+  }
+});
+
+test('finding a release without authorization', function (assert) {
```

Comment:
I don't think there is an existing example of either, but they're both something I want to do moving forward. Basically, just use assert.expect like below:

```
test('finding a release without authorization', function (assert) {
 assert.expect(18);

  server.get('/repos/user1/repository1/releases/1', () => {
    return [200, {}, Factory.build('release')];
  });

  return run(() => {
    return store.queryRecord('githubRelease', { repo: 'user1/repository1', releaseId: '1' }).then((release) => {
      assertGithubReleaseOk(assert, release);
      assert.equal(store.peekAll('githubRelease').get('length'), 1);
      assert.equal(server.handledRequests.length, 1);
      assert.equal(server.handledRequests[0].requestHeaders.Authorization, undefined);
    });
  });
});
```

An example of a test you wouldn't use this is where the assertions are done synchronously:

```
test('it serializes records', function (assert) {
  let record = this.subject();

  let serializedRecord = record.serialize();

  assert.ok(serializedRecord);
});
```

Basically, assert.expect allows us to have confidence that all asynchronous assertions were run, because otherwise you can get a passing test even if some of the assertions are never run due to a failure prior to the callback being executed.